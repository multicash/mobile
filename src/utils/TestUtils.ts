import fs from 'fs'
import path from 'path'
import { compileVueToRn } from 'vue-native-scripts'

export const getTestComponent = (component: string): any => {
  const data = fs.readFileSync(path.join(__dirname, '..', component), 'utf8')

  return compileVueToRn(data).output
}
