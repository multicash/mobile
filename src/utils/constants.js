export default {
  priceApi: 'https://api.vergecurrency.network/price/api/v1/price',
  ipApi: 'https://api.vergecurrency.network/price/api/v1/ip/',
  vwsApi: 'https://api.vergecurrency.network/vws/api',
  bnApi: 'https://api.vergecurrency.network/node/api/XVG',
  explorer: 'https://verge-blockchain.info',
  termsOfUse: 'https://vergecurrency.com/wallets/terms',
  feePerKb: 100000,
  satoshiDivider: 1000000,
  decimalPerSatoshi: 6,
  defaultCurrencyCode: 'USD',
  paperKeyLength: 12
}
