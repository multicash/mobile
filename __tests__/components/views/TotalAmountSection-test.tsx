import React from 'react'
import renderer from 'react-test-renderer'
import { getTestComponent } from '@/utils/TestUtils'

const TotalAmountSection = getTestComponent('components/views/TotalAmountSection.vue')

describe('Render total amount section component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TotalAmountSection />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
