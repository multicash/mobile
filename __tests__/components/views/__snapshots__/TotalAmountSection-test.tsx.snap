// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render total amount section component renders correctly 1`] = `
<import __react__vue__Vue, { observer as __react__vue__observer } from 'vue-native-core'
import __react__vue__ReactNative from 'react-native'
import __react__vue__PropType from 'prop-types'
import { buildNativeComponent as __react__vue__buildNativeComponent } from 'vue-native-helper'
import { bindNativeClass as __react__vue__bindClass,bindNativeStyle as __react__vue__bindStyle,mergeNativeStyleAndNativeClass as __react__vue__mergeNativeStyleAndNativeClass,mergeProps as __react__vue__mergeProps } from 'vue-native-helper'
import {
      createElement as __react__vue__createElement,
      Component as __react__vue__Component
    } from 'react'

import { Platform } from 'react-native'
const __react__vue__options =  {
  name: 'TotalAmountSection',

  computed: {
    styles () {
      return stylesStore(this.isDarkScheme)
    },

    totalAmount () {
      return this.wallets.map(wallet => wallet.totalAmount).reduce((a, b) => a + b, 0)
    }
  }
}

const stylesStore = (isDarkScheme) => {
  return {
    container: {
      flexGrow: 1,
      justifyContent: 'center',
      paddingHorizontal: 30
    },

    totalAmountTitle: {
      color: '#e1e1e1'
    },

    totalAmount: {
      color: '#ffffff',
      width: '100%',
      fontSize: 35,
      fontWeight: Platform.OS === 'ios' ? '900' : 'bold'
    },

    totalFiatAmount: {
      color: '#8de3ff',
      width: '100%',
      fontSize: 15,
      fontWeight: '600'
    }
  }
}

const __react__vue__render = function render(vm) {
    return __react__vue__createElement(vm.$options.components['View'], __react__vue__mergeProps.call(this, this.props.__react__vue__nativeEvents, {
        ref: ref => {
            this.setRootRef(ref);
            this.props['__react__vue__setRef'] && this.props['__react__vue__setRef'](ref);
        },
        style: __react__vue__mergeNativeStyleAndNativeClass(__react__vue__bindClass.call(this, {
            parentClass: this.props.style
        }), __react__vue__bindStyle(vm['styles'].container, undefined, undefined))
    }), __react__vue__createElement(vm.$options.components['Text'], {
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].totalAmountTitle, undefined, undefined))
    }, "Total Value"), __react__vue__createElement(vm.$options.components['Money'], {
        crypto: true,
        amount: vm['totalAmount'],
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].totalAmount, undefined, undefined))
    }), __react__vue__createElement(vm.$options.components['Money'], {
        approximately: true,
        convert: true,
        amount: vm['totalAmount'],
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].totalFiatAmount, undefined, undefined))
    }));
};

const __react__vue__css = {}

const __react__vue__ComponentBuilded = __react__vue__buildNativeComponent(__react__vue__render, __react__vue__options, {Component: __react__vue__Component, PropTypes: __react__vue__PropType, Vue: __react__vue__Vue, ReactNative: __react__vue__ReactNative, css: __react__vue__css})

export default __react__vue__observer(__react__vue__ComponentBuilded) />
`;
